### PR TITLE
Switched the TimerAlarms to use the RTC on the nrf51dk

### PR DESCRIPTION
**Pull request by KMarshland**
Previously, it was using a high frequency clock. This switches it over to use the lower power RTC.

### CODE REVIEW

**Comment by alevy** [comment]
The build failed because of formatting issues. You should be able to fix them by simply running: `make fmt`

File: boards/nrf51dk/src/main.rs [hunk 1]
```diff
@@ -163,16 +163,17 @@ pub unsafe fn reset_handler() {
     // as an HIL Alarm. Timer 0 has some special functionality for the BLE transciever,
     // so is reserved for that use. This should be rewritten to use the RTC (off the
     // low frequency clock) for lower power.
```
Comment (alevy) [inline boards/nrf51dk/src/main.rs]: I think we can get rid of this comment

File: chips/nrf51/src/rtc.rs [hunk 2]
```diff
@@ -94,23 +105,33 @@ impl Rtc {
...
+        rtc1().intenclr.set(COMPARE0_EVENT);
     }
 
     fn set_alarm(&self, tics: u32) {
+        //Similarly to the disable function, here we don't restart the timer
+        //Instead, we just listen for it again
         rtc1().cc[0].set(tics);
         rtc1().intenset.set(COMPARE0_EVENT);
-        self.start();
+        nvic::clear_pending(NvicIdx::RTC1);
```
Comment (alevy) [inline chips/nrf51/src/rtc.rs]: In general we shouldn't have to explicitly clear the pending NVIC, it should be cleared by us handling the event (this seems to be in several places)

File: kernel/src/hil/alarm.rs [hunk 3]
```diff
@@ -11,13 +11,27 @@ impl Frequency for Freq16MHz {
...
+}
+
 pub struct Freq16KHz;
 impl Frequency for Freq16KHz {
     fn frequency() -> u32 {
         16000
     }
 }
 
+pub struct Freq10KHz;
```
Comment (alevy) [inline kernel/src/hil/alarm.rs]: Since this is not actually used, please remove

**Comment by KMarshland** [comment]
Running `make fmt` gives me the following error: 
```
./tools/run_cargo_fmt.sh
find: illegal option -- p
usage: find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
       find [-H | -L | -P] [-EXdsx] -f path [path ...] [expression]
```

**Comment by ppannuto** [comment]
Sigh.. this is because you're on OS X which uses BSD tools instead of GNU
tools out of the box.

You can fix this for yourself immediately via:
brew install findutils --with-default-names

which will replace the default find binary. (Or w/out that flag it will
install as "gfind")

We should probably do something more permanent to fix make format to be
cross-platform though.

**Comment by alevy** [comment]
@KMarshland for now you can just run `cargo fmt` from each of the offending crates

**Pull request merged**